src/shared-validator.cc

```diff
@@ -48,15 +48,17 @@ Result SharedValidator::OnFuncType(const Location& loc,
                                    Index param_count,
                                    const Type* param_types,
                                    Index result_count,
-                                   const Type* result_types) {
+                                   const Type* result_types,
+                                   Index type_index) {
   Result result = Result::Ok;
   if (!options_.features.multi_value_enabled() && result_count > 1) {
     result |=
         PrintError(loc, "multiple result values not currently supported.");
   }
-  func_types_.emplace(num_types_++,
-                      FuncType{ToTypeVector(param_count, param_types),
-                               ToTypeVector(result_count, result_types)});
+  func_types_.emplace(
+      num_types_++,
+      FuncType{ToTypeVector(param_count, param_types),
+               ToTypeVector(result_count, result_types), type_index});
```

Comment:
I assume this formatting is what clang-format recommends?

Reply:
yep